code/modules/food_and_drink/snacks.dm

```diff
@@ -207,7 +207,60 @@
+	var/list/foodwhitelist
+	var/foodprob = 80
+
+	proc/filter_food(food_type)
+		if (ispath(food_type) in src.blacklist)
+			return FALSE
+		.= TRUE
+	proc/randomize_pizza(var/foodprob,var/list/blacklist,var/list/altfoodlist,var/list/foodwhitelist)
+		var/obj/item/reagent_containers/food/snacks/ingredient/pizza3/pizzabase = new /obj/item/reagent_containers/food/snacks/ingredient/pizza3
+		var/num_ingredients =  pick(16;1, 4;2, 1;3)
+
+		var/snack_type
+		var/ingredient_quality = 0.6
+		for (var/i = 1, i <= num_ingredients, i++)
+			if (prob(foodprob) && foodwhitelist)
+				snack_type = pick(foodwhitelist)
+			else if (prob(foodprob) && !foodwhitelist?.len > 0)
+				snack_type =  pick(filtered_concrete_typesof(/obj/item/reagent_containers/food/snacks, .proc/filter_food))
+			else if (altfoodlist?.len > 0)
+				snack_type = pick(altfoodlist)
+			else
+				snack_type = /obj/item/reagent_containers/food/snacks/ingredient/cheese
+			var/obj/item/reagent_containers/R = new snack_type
+			pizzabase.add_ingredient(R, null)
+			ingredient_quality += R.quality
+		var/datum/cookingrecipe/pizza/recipe = new
+		recipe.cook_pizza(pizzabase,src)
+		var/toppingstext = pizzabase.toppingstext
+		src.desc = "A pizza with [toppingstext] toppings. Looks...[pick("disturbing","interesting","awful","grody","kinda good","alright", "like a pizza","cursed","perfect","cruel","bland","old","unholy","squishy","wonderful","organic","fancy","traditional","rustic","experimental","enchanting","eldritch","horrific","palatable","questionable ","unnatural","forbidden","amateur","expensive","unweildy","alive","fake","broken","famous","blessed","tepid","round","cool","cute","radical","stylish","dry","moist","counterfeit","illegal","sour","transparent","used","digital","unhealthy","greasy","chewy","thin","cold","infectious","spicy")][pick("?",".","!")]"
```

Comment:
Disagree with the description being positive/negative having no correlation with the quality of the pizza.